### DISCUSSION TITLE
Automatically hides panel that has no data

### DISCUSSION THREAD

**Discussion by kevisong** [grafana/grafana#37969] [Dashboards, Panels & Visualizations]
Is it possible to add a setting in dashboard that automatically hides panel that has no data.

It can effectively filter out panel with no data which is generated by repeat option.

**Comment by pkolyvas** [comment]
Can you tell us a little more about your workflow? I'm particularly interested in a workflow that regularly generates/creates panels which don't have any data and aren't expected to?

**Reply by kevisong** [reply]
Absolutely. 

There are two scenarios that we want to hide the no data panels.

# Scenario 1

We have a dashboard that shows time series metrics of some HTTP endpoints(using repeat option) with anomaly details table on the bottom. Like this: 

![image](https://user-images.githubusercontent.com/39701004/130002198-de9eb7cb-b4e4-4481-98c5-04ccd5306e56.png)

In some cases, we will disable monitor data stream for one endpoint, let's say endpoint number 3(column number 3). which will result in no data in that column. OK, RTT and Success rate all showing no data. The no data panel will still be there because it's repeated based on another datasource that contains all endpoint names.

In this case, is it better to have an option to hide these no data panels so the other panels can be expanded larger.

# Scenario 2

Per above dashboard. When all endpoints are OK, the anomaly details will have no data, which takes up a lot of space
![image](https://user-images.githubusercontent.com/39701004/130002438-ad0924ba-6f5e-4bcc-a204-2aa87398514f.png)
If this table can be automatically hided (when no data). Panels that positioned under the table can be automatically moved up without wasting a huge chunk of space taken by the empty table.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
